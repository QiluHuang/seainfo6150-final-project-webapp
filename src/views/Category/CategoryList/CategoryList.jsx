import React, { useEffect, useState } from 'react';
import styles from './CategoryList.module.css';
import { Switch, Route, Link, useRouteMatch } from "react-router-dom";
import DishDetail from '../../DishDetail/DishDetail';
import CategoryListItem from '../CategoryListItem/CategoryListItem.jsx';



const CategoryList = (props) => {
    var cateID=props.categoryID;

    // console.log("props: " + props);

    return (
        <div className={styles.container}>
            <div className={styles.title}>
                <h3>{cateID}</h3>
            </div>
            <div>
                <div className={styles.menuList_container}>
                    {props.dishes.filter(dishes => dishes.categoryID==cateID).map((dish, index) => (
                            // console.log("dish: " + dish),
                            <li key={index}>
                                    <CategoryListItem dish={dish} key={dish.slug} />
                            </li>
                        ))}
                </div>
            </div>
            <div>
                
            </div>
        </div>
    );
}

export default CategoryList;