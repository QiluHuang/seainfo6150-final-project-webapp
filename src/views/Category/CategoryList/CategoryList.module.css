
  
.title {
    font-family: 'Times New Roman', Times, serif;
    font-size: 1.5em;
    text-align: center;
    text-decoration: underline;
}
  
.menuList_container {
      display: grid;
      grid-template-columns: repeat(4, 1fr);
      gap: 10px;
      font-family: Georgia, 'Times New Roman', Times, serif;
      list-style-type: none;
      padding: 10px;
}