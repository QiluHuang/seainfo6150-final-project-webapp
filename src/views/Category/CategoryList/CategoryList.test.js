import React from 'react';
import { render } from "@testing-library/react";
import CategoryList from './CategoryList';

const dish = [
   {
        slug: "0101-chicken-with-sweet-cream-corn-soup",
        name: "Chicken With Sweet Cream Corn Soup",
        categoryID : "soup",
        category: "Soup",
        cover: true,
        text: "It’s full of protein and veggies and it’s naturally gluten-free, and it is sooo comforting and delicious.ter.",
        ingredients: "Chicken, Corn, Ground ginger and garlic powder, Green Onions, Carrots, Cornstarch, Eggs, Toasted sesame oil, Salt and pepper",
        image: {
            _url: "https://imageresizer.static9.net.au/4X0xzcYzOVZeMKP7lsvoNmdEHz4=/1274x717/smart/https%3A%2F%2Fprod.static9.net.au%2F_%2Fmedia%2FImages%2FKitchen%2F2011%2F09%2F22%2F11%2F34%2Fchickensoup.jpg",
            _height: "282",
            _width: "501"
        },
        price: "$9.99"
    }   
];

describe("CategoryList tests", () => {
    it("renders correctly", () => {
      const { container } = render(<CategoryList categoryID={dish.categoryID} 
        dishes={dish}/> );
      expect(container).toMatchSnapshot();
    });
});
  