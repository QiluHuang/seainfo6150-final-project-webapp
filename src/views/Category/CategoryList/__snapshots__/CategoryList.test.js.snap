// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryList tests renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="title"
    >
      <h3 />
    </div>
    <div>
      <div
        class="menuList_container"
      />
    </div>
    <div />
  </div>
</div>
`;
